### PR TITLE
chore(docs): add BOM install instructions for gradle

### PR DESCRIPTION
**Pull request by elharo**
@chingor13 I don't see an easy way to find the latest version of libraries-bom in the template. I think I need to add a build.gradle in samples that renovatebot could see?

### CODE REVIEW

**Comment by chingor13** [comment]
We collect some common metadata in the [java specific templates helper](https://github.com/googleapis/synthtool/blob/6b026e1443948dcfc0b9e3289c85e940eb70f694/synthtool/languages/java.py#L346-L348). 

It should be available as `metadata['latest_bom_version']`.

**Comment by elharo** [comment]
Thanks. That's exactly what I needed. It should be ready for review now.

**Review by SurferJeffAtGoogle — COMMENTED** [review]
LGTM.  I leave it to @chingor13 to approve.

File: synthtool/gcp/templates/java_library/README.md [hunk 1]
```diff
@@ -38,10 +38,20 @@ If you are using Maven, add this to your pom.xml file:
 {% endif -%}
 ```
 
-If you are using Gradle, add this to your dependencies
+{% if 'snippets' in metadata and metadata['snippets'][metadata['repo']['name'] + '_install_with_bom'] -%}
+If you are using Gradle 5.x or later, add this to your dependencies
+```Groovy
+implementation platform('com.google.cloud:libraries-bom:metadata['latest_bom_version']')
```
Comment (chingor13) [inline synthtool/gcp/templates/java_library/README.md:44]:
```suggestion
implementation platform('com.google.cloud:libraries-bom:{{metadata['latest_bom_version']}}')
```

Need to use jinja template variable tags

**Review by chingor13 — CHANGES_REQUESTED** [review]

**Review by elharo — COMMENTED** [review]
done

**Review by chingor13 — APPROVED** [review]

**Pull request merged**